third_party/terraform/utils/iam_iap_tunnel.go.erb

```diff
@@ -0,0 +1,188 @@
+<% autogen_exception -%>
+package google
+<% unless version == 'ga' -%>
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/errwrap"
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/cloudresourcemanager/v1"
+	iap "google.golang.org/api/iap/v1beta1"
+)
+
+var IamIapTunnelSchema = map[string]*schema.Schema{
+	"instance": {
+		Type:     schema.TypeString,
+		Required: true,
+		ForceNew: true,
+	},
+
+	"project": {
+		Type:     schema.TypeString,
+		Optional: true,
+		Computed: true,
+		ForceNew: true,
+	},
+
+	"zone": {
+		Type:     schema.TypeString,
+		Optional: true,
+		Computed: true,
+		ForceNew: true,
+	},
+}
+
+type IapTunnelIamUpdater struct {
+	project       string
+	zone          string
+	resourceId    string
```

Comment:
iam_compute_subnetworks.go uses resourceId, which is what I used to base a lot of this work on.

Reply:
Fair enough! I think my preference is for `instance`, but feel free to use `resourceId` if that's yours.

Reply:
I actually like `instance` better too. resourceId was mostly out of staying consistent/afraid to change what others had done in case it had other implications.